Runtime/Event/BehaviourEnabledObserver.cs

```diff
@@ -1,53 +1,136 @@
         /// </summary>
         [DocumentedByXml]
         public UnityEvent ActiveAndEnabled = new UnityEvent();
 
+        /// <summary>
+        /// A reference to the started routine.
+        /// </summary>
+        protected Coroutine behaviourCheckRoutine;
+        /// <summary>
+        /// Delays the <see cref="behaviourCheckRoutine"/> by <see cref="CheckDelay"/> seconds.
+        /// </summary>
+        protected WaitForSeconds checkDelayYield;
+
+        /// <summary>
+        /// Initiates the check of the <see cref="Behaviours"/> state if no existing check is already running.
+        /// </summary>
+        public virtual void BeginCheck()
+        {
+            if (behaviourCheckRoutine == null)
+            {
+                behaviourCheckRoutine = StartCoroutine(Check());
+            }
+        }
+
+        /// <summary>
+        /// Cancels any running check of the <see cref="Behaviours"/> state.
+        /// </summary>
+        public virtual void EndCheck()
+        {
+            if (behaviourCheckRoutine != null)
+            {
+                StopCoroutine(behaviourCheckRoutine);
+            }
+            behaviourCheckRoutine = null;
+        }
+
         protected virtual void OnEnable()
         {
-            InvokeRepeating(nameof(Check), MinimumRepeatRate, MinimumRepeatRate);
+            BeginCheck();
         }
 
         protected virtual void OnDisable()
         {
-            CancelInvoke(nameof(Check));
+            EndCheck();
+        }
+
+        /// <summary>
+        /// Checks to see if the <see cref="Behaviours"/> specified have been enabled in the scene.
+        /// </summary>
+        /// <returns>An Enumerator to manage the running of the Coroutine.</returns>
+        protected virtual IEnumerator Check()
+        {
+            float normalizedTime = 0;
+            checkDelayYield = new WaitForSeconds(CheckDelay);
+            while (normalizedTime <= 1f)
+            {
+                if (AreBehavioursEnabled())
+                {
+                    break;
+                }
+                normalizedTime += Time.deltaTime / MaximumRunTime;
+                yield return checkDelayYield;
+            }
+            behaviourCheckRoutine = null;
         }
 
         /// <summary>
-        /// Checks whether all <see cref="behaviours"/> are <see cref="Behaviour.isActiveAndEnabled"/> and emits <see cref="ActiveAndEnabled"/> if they are.
+        /// Checks whether all <see cref="Behaviours"/> are <see cref="Behaviour.isActiveAndEnabled"/> and emits <see cref="ActiveAndEnabled"/> if they are.
         /// </summary>
-        protected virtual void Check()
+        /// <returns>Whether all <see cref="Behaviours"/> are active and enabled.</returns>
+        protected virtual bool AreBehavioursEnabled()
         {
-            if (!behaviours.TrueForAll(behaviour => behaviour.isActiveAndEnabled))
+            if (Behaviours == null || Behaviours.NonSubscribableElements.Count == 0)
```

Comment:
`NonSubscribableElements` here is correct, but brings up a point: Shouldn't (almost) all the (renamed) usages of `SubscribableElements` be `NonSubscribableElements` instead? I think we should ensure the correct ones are used... Maybe we just create a new ticket for it and come back to it another time.

Reply:
Using the `NonSubscribableElements` would also allow the tests that were updated to use [UnityTest] be able to go back to just using [Test] in many cases.

However, this is probably a large enough change to make it future work and not just add it to this PR.

Reply:
I created https://github.com/ExtendRealityLtd/Zinnia.Unity/issues/326 for this.